blocks/library/quote/index.js

```diff
@@ -63,9 +89,13 @@ registerBlockType( 'core/quote', {
 				blocks: [ 'core/heading' ],
 				transform: ( { value, citation, ...attrs } ) => {
 					const isMultiParagraph = Array.isArray( value ) && isObject( value[ 0 ] ) && value[ 0 ].type === 'p';
+					const headingElement = isMultiParagraph ? value[ 0 ] : value;
+					const headingContent = isObject( headingElement ) && value[ 0 ].type === 'p'
```

Comment:
Similarly here, sometimes we seem to check for string, other times object. Nitpicking now though. :D